h/search/query.py

```diff
@@ -3,127 +3,99 @@
-        aggregations = {a.key: a(params) for a in self.aggregations}
-        filters = [f for f in filters if f is not None]
-        matchers = [m for m in matchers if m is not None]
+    Sorts annotations by sort (the key to sort by)
+    and the order (the order in which to sort by).
+    """
 
-        # Remaining parameters are added as straightforward key-value matchers
-        for key, value in params.items():
-            matchers.append({"match": {key: value}})
-
-        # See https://www.elastic.co/guide/en/elasticsearch/reference/6.2/query-dsl-filtered-query.html
-        query = {
-            "bool": {
-                "filter": filters,
-                "must": matchers,
-            },
-        }
-
-        return {
-            "from": p_from,
-            "size": p_size,
-            "sort": p_sort,
-            "query": query,
-            "aggs": aggregations,
-        }
-
-
-def extract_offset(params):
-    try:
-        val = int(params.pop("offset"))
-        val = min(val, OFFSET_MAX)
-        if val < 0:
-            raise ValueError
-    except (ValueError, KeyError):
-        return 0
-    else:
-        return val
-
-
-def extract_limit(params):
-    try:
-        val = int(params.pop("limit"))
-        val = min(val, LIMIT_MAX)
-        if val < 0:
-            raise ValueError
-    except (ValueError, KeyError):
-        return LIMIT_DEFAULT
-    else:
-        return val
-
-
-def extract_sort(params):
-    sort_by = params.pop("sort", "updated")
-    # Sorting must be done on non-analyzed fields.
-    if sort_by == "user":
-        sort_by = "user_raw"
-    return [{
-        sort_by: {
-            "order": params.pop("order", "desc"),
-
-            # `unmapped_type` causes unknown fields specified as arguments to
-            # `sort` behave as if all documents contained empty values of the
-            # given type. Without this, specifying eg. `sort=foobar` throws
-            # an exception.
-            #
-            # We use the field type `boolean` to assist with migration because
-            # that exists in both ES 1 and ES 6.
-            "unmapped_type": "boolean",
-        }
-    }]
+    def __call__(self, search, params):
+        sort_by = params.pop("sort", "updated")
+        # Sorting must be done on non-analyzed fields.
+        if sort_by == "user":
+            sort_by = "user_raw"
+        return [{
+            sort_by: {
+                "order": params.pop("order", "desc"),
+
+                # `unmapped_type` causes unknown fields specified as arguments to
+                # `sort` behave as if all documents contained empty values of the
+                # given type. Without this, specifying eg. `sort=foobar` throws
+                # an exception.
+                #
+                # We use the field type `boolean` to assist with migration because
+                # that exists in both ES 1 and ES 6.
+                "unmapped_type": "boolean",
```

Comment:
Don't change it in this PR, but I note that we could drop this now since we are validating the `sort` parameter.

Reply:
^ I also noted this but then I thought we need to be careful about this cause we are validating it in the api, but I'm not sure about the other places where we call search? I don't think we use sort in those places/let it slip through but I'll need to double check before removing.